frame/support/procedural/src/storage/genesis_config/builder_def.rs

```diff
@@ -120,6 +120,24 @@ impl BuilderDef {
 							});
 						}}
 					},
+					StorageLineTypeDef::NMap(map) => {
+						let keygen = map.to_keygen_struct(scrate);
+						let key_tuple = map.to_key_tuple();
+						let key_arg = if map.keys.len() == 1 {
+							quote!((k,))
```

Comment:
This is the only place where it is correct, because the `k` here is not bounded by `KArg`, hence it'll need to `EncodeLikeTuple<(Key1,)>`, i.e. `k` must be a 1-element tuple.